pennylane/ops/qubit/qchem_ops.py

```diff
@@ -464,3 +464,146 @@ def _matrix(cls, *params):
+                            0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1 & 0 & 0 & 0 & 0 & 0\\
+                            0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & c & 0 & 0 & - s & 0\\
+                            0 & 0 & 0 & s^{2} & 0 & 0 & c s & 0 & 0 & c s & 0 & 0 & c^{2} & 0 & 0 & 0\\
+                            0 & 0 & 0 & 0 & 0 & 0 & 0 & s & 0 & 0 & 0 & 0 & 0 & c & 0 & 0\\
+                            0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & s & 0 & 0 & c & 0\\
+                            0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 0 & 1
+                        \end{bmatrix}.
+
+    This operation performs the spatial orbital Givens rotation :math:`|\phi_{0}\rangle = c|\phi_{0}\rangle - s|\phi_{1}\rangle`
+    and :math:`|\phi_{1}\rangle = s|\phi_{0}\rangle + c|\phi_{1}\rangle`.
+
+    **Details:**
+
+    * Number of wires: 4
+    * Number of parameters: 1
+    * Gradient recipe: The ``QuantumNumberPreservingOR`` operator satisfies a four-term parameter-shift rule
+      (see Appendix F, https://arxiv.org/abs/2104.05695)
+
+    Args:
+        varphi (float): rotation angle :math:`\varphi`
+        wires (Sequence[int]): the wires the operation acts on
+
+    **Example**
+
+    .. code-block::
+
+        dev = qml.device('default.qubit', wires=4)
+
+        @qml.qnode(dev)
+        def circuit(varphi):
+            qml.QuantumNumberPreservingOR(varphi, wires=[0, 1, 2, 3])
+            return qml.state()
+
+        circuit(0.1)
+    """
+
+    num_params = 1
+    num_wires = 4
+    par_domain = "R"
+    grad_method = "A"
+    grad_recipe = four_term_grad_recipe
+    generator = [
+        np.array(
+            [
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, -1j, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, -1j, 0, 0, -1j, 0, 0, 0, 0, 0, 0],
+                [0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0],
+                [0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, -1j, 0],
+                [0, 0, 0, 0, 0, 0, 1j, 0, 0, 1j, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 1j, 0, 0, 0, 0, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1j, 0, 0, 0, 0],
+                [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+            ]
+        ),
+        -1 / 2,
+    ]
+
+    @classmethod
+    def _matrix(cls, *params):
+        theta = params[0]
+        c = math.cos(theta / 2)
+        s = math.sin(theta / 2)
+
+        return np.array(
```

Comment:
Please see my comments regarding the signs.

Reply:
Done.